### PR TITLE
Bug 1712293: frontend/public: use working set bytes for memory metrics

### PR DESCRIPTION
**Pull request by s-urbaniak**

### CODE REVIEW

File: frontend/public/components/build.tsx [hunk 1]
```diff
@@ -122,7 +122,7 @@ const BuildGraphs = requirePrometheus(({build}) => {
           title="Memory Usage"
           humanize={humanizeDecimalBytes}
           namespace={namespace}
-          query={`pod_name:container_memory_usage_bytes:sum{pod_name='${podName}',container_name='',namespace='${namespace}'}`}
+          query={`sum(container_memory_working_set_bytes{pod_name='${podName}',namespace='${namespace}',container_name!='POD',container_name!='',pod_name!=''}) BY (pod_name, namespace)`}
```
[the remark this answers is not in the document]
Comment (brancz) [reply frontend/public/components/build.tsx]: That’s correct, we need to change it to the pod’s cgroup.

[the remark this answers is not in the document]
Comment (s-urbaniak) [reply frontend/public/components/build.tsx]: ah good catch, a classic copy-paste :+1:

**Comment by brancz** [comment]
Mostly looks good, except I think this query also needs to use working set bytes: https://github.com/openshift/console/blob/36615d459c9840dd621a82b83188d4094f0585a0/frontend/public/components/namespace.jsx#L273

File: frontend/public/components/namespace.jsx [hunk 2]
```diff
@@ -270,7 +270,7 @@ export const TopPodsBarChart = ({ns}) => (
   <Bar
     title="Memory Usage by Pod (Top 10)"
     namespace={ns.metadata.name}
-    query={`sort_desc(topk(10, sum by (pod_name)(container_memory_usage_bytes{container_name!="POD",container_name!="",pod_name!="", namespace="${ns.metadata.name}"})))`}
+    query={`sort_desc(topk(10, sum by (pod_name)(container_memory_working_set_bytes{container_name!="POD",container_name!="",pod_name!="", namespace="${ns.metadata.name}"})))`}
```
Comment (brancz) [inline frontend/public/components/namespace.jsx]: we're looking for the topk pod right? in that case we should filter with `container=""`, so we get the pod cgroup hierarchy

**Comment by s-urbaniak** [comment]
@smarterclayton @brancz i hopefully addressed the comments, ptal.

**Comment by brancz** [comment]
/lgtm

but let's verify once by launching a cluster, feel free to remove hold once verified

/hold

**Comment by kyoto** [comment]
@s-urbaniak Needs rebase due to https://github.com/openshift/console/pull/2483

**Comment by s-urbaniak** [comment]
i still have some issue with the overview dashboard pane not showing memory metrics. The executed query looks fine:

```
sort_desc(sum(container_memory_working_set_bytes{container="",pod!=""}) BY (pod, namespace))
```

image

**Comment by spadgett** [comment]
cc @rawagner

**Comment by rawagner** [comment]
@s-urbaniak the issue with Top Consumers card is solved, right ? I see that you changed https://github.com/openshift/console/pull/2486/files#diff-163991d4c119db922a729100ce2bea7fR35

**Comment by s-urbaniak** [comment]
@rawagner yes I resolved it:
image

**Comment by s-urbaniak** [comment]
fyi: I will submit a follow-up PR factoring several node recording rules too as https://github.com/openshift/cluster-monitoring-operator/pull/456 replaces a lot of them as discussed out of band with @spadgett.

**Comment by s-urbaniak** [comment]
@rawagner @spadgett hmmm, do you mind to have a peek at the aws console test failure? :thinking: i can't think why this PR could've broken it (just changed queries) :thinking:

**Comment by paulfantom** [comment]
@s-urbaniak from what I see this is not related to console, but it happens globally in CI. We are experiencing the same errors at https://github.com/openshift/cluster-monitoring-operator/pull/456

**Comment by s-urbaniak** [comment]
@paulfantom ah, now i see too the logs, thanks for the heads-up :+1:

**Pull request merged**